src/bun.js/bindings/BunProcess.cpp

```diff
@@ -1784,17 +1771,57 @@ JSC_DEFINE_HOST_FUNCTION(Process_functionAssert, (JSGlobalObject * globalObject,
-        throwScope.throwException(globalObject, createError(globalObject, String("process.binding(\"" str "\") is not implemented in Bun. If that breaks something, please file an issue and include a reproducible code sample."_s))); \
-        return JSValue::encode(JSValue {});                                                                                                                                                                                             \
-    }
+inline JSC::EncodedJSValue createProcessBindingNotImplementedProxy(
+    JSC::JSGlobalObject* globalObject,
+    JSC::VM& vm,
+    WTF::String handlerGetTrapCode)
+{
+    auto nullProtoStruct = globalObject->nullPrototypeObjectStructure();
+    auto&& origin = SourceOrigin(WTF::URL());
+    auto target = JSC::constructEmptyObject(vm, nullProtoStruct);
+    auto handler = JSC::constructEmptyObject(vm, nullProtoStruct);
+    SourceCode handlerGetTrapSourceCode
+        = JSC::makeSource(
+            handlerGetTrapCode,
+            origin,
+            JSC::SourceTaintedOrigin::Untainted);
+    JSFunction* func
+        = JSFunction::create(
+            vm,
+            JSC::createBuiltinExecutable(
+                vm,
+                handlerGetTrapSourceCode,
+                Identifier(),
+                ImplementationVisibility::Private,
+                ConstructorKind::None,
+                ConstructAbility::CannotConstruct,
+                InlineAttribute::Always)
+                ->link(vm, nullptr, handlerGetTrapSourceCode),
+            static_cast<JSC::JSGlobalObject*>(globalObject));
+    handler->putDirect(vm, Identifier::fromString(vm, "get"_s), func, 0);
+    return JSValue::encode(ProxyObject::create(globalObject, target, handler));
+}
+#define PROCESS_BINDING_NOT_IMPLEMENTED_1_ARGS(moduleName)                                                                                                                                            \
+    {                                                                                                                                                                                                 \
+        auto handlerGetTrapCode = String(                                                                                                                                                             \
+            "(function (_target,propertyKey){"                                                                                                                                                        \
+            "throw new Error(`process.binding(\"" moduleName "\").${propertyKey} is not implemented in Bun. If that breaks something, please file an issue and include a reproducible code sample.`)" \
+            "})"_s);                                                                                                                                                                                  \
```

Comment:
Creates the handler get trap code using the macro to avoid multiple string object creations (like `jsString()` to concat them).